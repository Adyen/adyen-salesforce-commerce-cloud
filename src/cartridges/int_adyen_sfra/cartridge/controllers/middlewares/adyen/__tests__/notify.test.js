/* eslint-disable global-require */

let req;
let res;
let notify;

beforeEach(() => {
  const { adyen } = require('../../index');
  notify = adyen.notify;
  jest.clearAllMocks();
  req = {};
  res = { render: jest.fn() };
});

afterEach(() => {
  jest.resetModules();
});

describe('Notify', () => {
  it('should render error when status is falsy', () => {
    const checkAuth = require('*/cartridge/scripts/checkNotificationAuth');
    checkAuth.check.mockImplementation(() => false);
    notify(req, res, jest.fn());
    expect(res.render).toHaveBeenCalledWith('/adyen/error');
  });
  it('should render notify when notification result is successful', () => {
    const handleNotify = require('*/cartridge/scripts/handleNotify');
    handleNotify.notify.mockImplementation(() => ({ success: true }));
    notify(req, res, jest.fn());
    expect(res.render).toHaveBeenCalledWith('/notify');
  });
  it('should render notifyError when notification result is not successful', () => {
    const handleNotify = require('*/cartridge/scripts/handleNotify');
    handleNotify.notify.mockImplementation(() => ({
      success: false,
      errorMessage: 'mocked_error_message',
    }));
    notify(req, res, jest.fn());
    expect(res.render.mock.calls).toMatchSnapshot();
  });
});
