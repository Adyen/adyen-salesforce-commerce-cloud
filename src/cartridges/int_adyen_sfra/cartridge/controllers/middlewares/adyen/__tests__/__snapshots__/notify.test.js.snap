// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notify should render notifyError when notification result is not successful 1`] = `
Array [
  Array [
    "/notifyError",
    Object {
      "errorMessage": "mocked_error_message",
    },
  ],
]
`;
